From e4ab952d639ff69e57fefa3e3855ee81d4d555fb Mon Sep 17 00:00:00 2001
From: Peter Hjalmarsson <xake@rymdraket.net>
Date: Tue, 5 Mar 2013 22:45:45 +0100
Subject: [PATCH 07/14] Add rootflags to switches we understand

Nearly every general documentation including the one in the
kernel tree and the scripts for GRUB2 all expects this flag
to work.

Signed-off-by: Peter Hjalmarsson <xake@rymdraket.net>
---
 defaults/linuxrc    | 24 +++++++++++++++++-------
 doc/genkernel.8.txt |  8 ++++++--
 2 files changed, 23 insertions(+), 9 deletions(-)

diff --git a/defaults/linuxrc b/defaults/linuxrc
index f1e20ce..703652d 100755
--- a/defaults/linuxrc
+++ b/defaults/linuxrc
@@ -37,7 +37,7 @@ CMDLINE=$(cat /proc/cmdline)
 # Scan CMDLINE for any specified real_root= or cdroot arguments
 FAKE_ROOT=''
 FAKE_INIT=''
-REAL_ROOTFLAGS=''
+FAKE_ROOTFLAGS=''
 INIT_OPTS=''
 ROOTFSTYPE='auto'
 CRYPT_SILENT=0
@@ -236,6 +236,9 @@ do
 		real_rootflags=*)
 			REAL_ROOTFLAGS=${x#*=}
 		;;
+		rootflags=*)
+			FAKE_ROOTFLAGS=${x#*=}
+		;;
 		rootfstype=*)
 			ROOTFSTYPE=${x#*=}
 		;;
@@ -267,13 +270,20 @@ do
 	esac
 done
 
-if [ -z "${REAL_ROOT}" -a \( "${CDROOT}" = '0' \)  -a \( "${FAKE_ROOT}" != "/dev/ram0" \) ]
-then
-	REAL_ROOT="${FAKE_ROOT}"
-fi
-if [ -z "${REAL_INIT}" -a \( "${CDROOT}" = '0' \)  -a \( "${FAKE_INIT}" != "/linuxrc" \) ]
+if [ \( "${CDROOT}" = '0' \) ]
 then
-	REAL_INIT="${FAKE_INIT}"
+	if [ -z "${REAL_ROOT}" -a \( "${FAKE_ROOT}" != "/dev/ram0" \) ]
+	then
+		REAL_ROOT="${FAKE_ROOT}"
+	fi
+	if [ -z "${REAL_INIT}" -a \( "${FAKE_INIT}" != "/linuxrc" \) ]
+	then
+		REAL_INIT="${FAKE_INIT}"
+	fi
+	if [ -z "${REAL_ROOTFLAGS}" ]
+	then
+		REAL_ROOTFLAGS="${FAKE_ROOTFLAGS}"
+	fi
 fi
 
 # Set variables based on the value of REAL_ROOT
diff --git a/doc/genkernel.8.txt b/doc/genkernel.8.txt
index 4691a98..86d7d53 100644
--- a/doc/genkernel.8.txt
+++ b/doc/genkernel.8.txt
@@ -545,9 +545,13 @@ which the ramdisk scripts would recognize.
 *aufs*::
     Enables support for AUFS2 (if available in the kernel).
 
-*real_rootflags*=<...>::
+*rootflags*=<...>::
     Additional flags to mount the real root system with.
-    For example *real_rootflags*=noatime would make "-o ro,noatime".
+    For example *rootflags*=noatime would make "-o ro,noatime".
+
+*real_rootflags*=<...>::
+    Legacy kernel parameter from kernel-2.4 initrd.
+    Does the same as *rootflags*=, which should be used in its place.
 
 *real_resume*=<...>::
 *resume*=<...>::
-- 
1.8.1.5

